include/deal.II/fe/mapping_q_cache.h

```diff
@@ -161,6 +167,38 @@ class MappingQCache : public MappingQGeneric<dim, spacedim>
              const Function<spacedim> &          transformation_function,
              const bool function_describes_relative_displacement);
 
+  /**
+   * Initialize the data cache of the active cells by a solution (specified by
+   * @p dof_handler and @p vector) that describes the absolute or
+   * relative position of each support point.
+   *
+   * @note By using this function for reinitialization, this class behaves like
+   *   MappingFEField (vector_describes_relative_displacement == false) or
+   *   MappingQEulerian (vector_describes_relative_displacement == true).
+   */
+  template <typename VectorType>
+  void
+  initialize(const Mapping<dim, spacedim> &   mapping,
+             const DoFHandler<dim, spacedim> &dof_handler,
+             const VectorType &               vector,
+             const bool vector_describes_relative_displacement);
```

Comment:
Do we want to add an optional `ComponentMask` argument to this function, in analogy to `MappingFEField`? This would allow to extract data from a `DoFHandler` that contains additional solution components.

Reply:
Let's do this in a follow-up PR. It does not look too complicated (component need to be linearized and we need some additional checks and a new test).